app/src/main/java/com/github/se/wanderpals/ui/screens/trip/map/Map.kt

```diff
@@ -0,0 +1,368 @@
+        cameraPositionState =
+            CameraPositionState(position = CameraPosition(finalLocation, 10f, 0f, 0f))) {
+
+          // display current location on the map
+          if (seeCurrentLocation) {
+            Marker(
+                state = MarkerState(position = currentLocation),
+                title = "Current Location",
+                icon = BitmapDescriptorFactory.fromResource(R.drawable.logo_position))
+          }
+
+          // display the marker on the map
+          listOfMarkers.forEach { markerState ->
+            Marker(
+                // Add a marker to the map
+                state = markerState,
+                title = "Click to Create Suggestions",
+                onInfoWindowClick = {
+                  oldNavActions.setVariablesLocation(
+                      GeoCords(markerState.position.latitude, markerState.position.longitude),
+                      placeData.placeAddress)
+                  oldNavActions.navigateTo(Route.CREATE_SUGGESTION)
+                })
+          }
+
+          // display all the stops on the map
+          stopList.forEach {
+            Marker( // Add a marker to the map
+                state = MarkerState(position = LatLng(it.geoCords.latitude, it.geoCords.longitude)),
+                title = it.title,
+                snippet = it.description,
+                icon = BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_AZURE))
+          }
+
+          // display all the suggestions on the map
+          suggestionList.forEach {
+            Marker( // Add a marker to the map
+                state = MarkerState(position = LatLng(it.geoCords.latitude, it.geoCords.longitude)),
+                title = it.title,
+                snippet = it.description,
+                icon = BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_GREEN))
+          }
+
+          // display all the users positions on the map
+          usersPositions.forEach {
+            Marker( // Add a marker to the map
+                state = MarkerState(position = it),
+                title = userNames[usersPositions.indexOf(it)],
+                icon = BitmapDescriptorFactory.fromResource(R.drawable.logo_position_other))
+          }
+        }
+    Switch(
+        modifier =
+            Modifier.testTag("switchButton")
+                .align(AbsoluteAlignment.BottomLeft)
+                .padding(horizontal = 16.dp, vertical = 60.dp),
+        checked = enabledSearchBar,
+        onCheckedChange = { enabledSearchBar = !enabledSearchBar })
+
+    Column(
+        Modifier.align(AbsoluteAlignment.BottomRight)
+            .padding(horizontal = 16.dp, vertical = 60.dp)) {
+          Button(
+              onClick = {
+                coroutineScope.launch(Dispatchers.IO) {
+                  mapManager.getLastLocation().let {
+                    if (it != null) {
+                      finalLocation = it
+                      mapViewModel.updateLastPosition(it)
```

Comment:
Its not recommended to have coroutines in the UI, you should try to move the couroutine blockinto a viewmodel, will make it cleaner, and more "correct". Which will also improve maintainability.